SwiftRuntimeLibrary/Makefile

```diff
@@ -0,0 +1,34 @@
+TOP=..
+include $(TOP)/common.mk
+
+configuration ?= debug
+BINDING_METADATA_IOS = SwiftRuntimeLibrary.iOS/GeneratedCode/BindingMetadata.iOS.cs
+BINDING_METADATA_MAC = SwiftRuntimeLibrary.Mac/GeneratedCode/BindingMetadata.MacOS.cs
+
+all: create-pinvokes
+
+clean-ios:
+	@echo "Cleaning binary files and generated code for iOS"
+	@rm -rf $(TOP)/SwiftRuntimeLibrary.iOS/bin/*
+	@rm -rf $(TOP)/SwiftRuntimeLibrary.iOS/GeneratedCode
+
+clean-mac:
+	@echo "Cleaning binary files and generated code for Mac"
+	@rm -rf $(TOP)/SwiftRuntimeLibrary.Mac/bin/*
+	@rm -rf $(TOP)/SwiftRuntimeLibrary.Mac/GeneratedCode
+
+clean: clean-ios clean-mac
+	
+create-pinvokes-ios:
```

Comment:
This is not the optimal way of writing a makefile, because the makefile will re-execute everything if you run `make` again, even if nothing changed.

The better way to do it is to identify the inputs and output of the task you want to accomplish. In this case it seems the output is `$(TOP)/$(BINDING_METADATA_IOS)`, but I'm not sure what the inputs are.

In any case, the format of the make rule would be something like this:

```make
$(TOP)/$(BINDING_METADATA_IOS): inputfile1 inputfile2 inputfileX
    @echo "Generating code for iOS..."
    @mkdir -p $(TOP)/SwiftRuntimeLibrary.iOS/GeneratedCode
    @/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono $(TOP)/type-o-matic/bin/Debug/type-o-matic.exe --swift-lib-path=$(TOP)/apple/build/Ninja-ReleaseAssert/swift-macosx-x86_64/lib/swift/ --platform=iphone --generate=csharp $(addprefix --namespace=, $(TYPE_O_MATIC_IOS_NAMESPACES)) --xamglue-framework=$(TOP)/swiftglue/bin/$(configuration)/iphone/FinalProduct/XamGlue.framework --output=$(TOP)/$(BINDING_METADATA_IOS)
    @echo "Generated C# binding metadata files for iOS at $(TOP)/$(BINDING_METADATA_IOS)"

$(TOP)/$(BINDING_METADATA_MAC): ...
    ...

create-pinvokes: $(TOP)/$(BINDING_METADATA_IOS) $(TOP)/$(BINDING_METADATA_MAC)
```

Reply:
@xxaxdxcxx please address this before we merge.

Reply:
@rolfbjarne @mandel-macaque addressed in latest commit :)